src/java/org/apache/cassandra/index/sai/disk/v2/sortedterms/SortedTermsReader.java

```diff
@@ -127,13 +127,29 @@ public long getPointId(@Nonnull ByteComparable term)
     }
 
     /**
-     * Returns the last point id (ordinal) of the target term or the next smaller if no exact match found.
-     * If reached the end of the terms file, returns <code>Long.MAX_VALUE</code>.
+     * Returns the point id (ordinal) of the target term or <code>Long.MAX_VALUE</code> if there is no such term.
+     * Complexity of this operation is O(log n).
+     *
+     * @param term target term to lookup
+     */
+    public long getExactPointId(@Nonnull ByteComparable term)
+    {
+        Preconditions.checkNotNull(term, "term null");
+        try (TrieTermsDictionaryReader reader = new TrieTermsDictionaryReader(termsTrie.instantiateRebufferer(), meta.trieFP))
+        {
+            long result = reader.exactMatch(term);
+            return result < 0 ? Long.MAX_VALUE : result;
+        }
+    }
```

Comment:
@jbellis - I'm pretty sure this is still `O(log n)`, but the `TrieTermsDictionaryReader` should be better in terms of object initialization. The `TrieRangeIterator` spends time building a stack, while the `TrieTermsDictionaryReader` only traverses the trie. Let me know what you think.

Reply:
Not my expertise but lgtm.